### PR TITLE
Alex krasn/fix render error on png files #244

### PR DESCRIPTION
**Pull request by alex-krasn**
On issue #244 
please run 'npm install' before testing

### CODE REVIEW

**Comment by alex-krasn** [comment]
@stew-ro Could you, please, share the project, so I can reproduce?

File: src/services/assetService.ts [hunk 1]
```diff
@@ -92,7 +113,7 @@ export class AssetService {
     }
 
     /**
-     * Get Asset Provider from project's source connction
+     * Get Asset Provider from project"s source connection
```
Comment (stew-ro) [inline src/services/assetService.ts:116]: should be '

Comment (alex-krasn) [reply src/services/assetService.ts]: fixed

File: src/services/assetService.ts [hunk 2]
```diff
@@ -84,6 +95,15 @@ export class AssetService {
         }
     }
 
+    public static getMimeType = async (uri: string) => {
+        const FileType = require("file-type/browser");
+        const response = await fetch(uri);
```
Comment (stew-ro) [inline src/services/assetService.ts:100]: We don't want to fetch each document twice

File: src/services/assetService.ts [hunk 3]
```diff
@@ -84,6 +95,15 @@ export class AssetService {
...
+        const fileType = await FileType.fromStream(response.body);
```
Comment (stew-ro) [inline src/services/assetService.ts:101]: it's not a stream

File: package.json [hunk 4]
```diff
@@ -19,6 +19,7 @@
         "bootstrap": "^4.4.1",
         "chart.js": "^2.9.3",
         "exif-js": "^2.3.0",
+        "file-type": "^14.3.0",
```
Comment (kunzms) [inline package.json:22]: add license declaration in ThirdPartyNotice.txt

File: src/services/assetService.ts [hunk 5]
```diff
@@ -16,6 +16,42 @@ import { strings, interpolate } from "../common/strings";
 import { sha256Hash } from "../common/crypto";
 import { toast } from "react-toastify";
 
+const supportedImageFormats = ["jpg", "jpeg", "png", "bmp", "tif", "tiff", "pdf"];
+
+interface IMime {
+    mime: string;
+    pattern: Array<number | undefined>;
+  }
```
Comment (kunzms) [inline src/services/assetService.ts:24]: correct the indent here

File: src/services/assetService.ts [hunk 6]
```diff
@@ -16,6 +16,42 @@ import { strings, interpolate } from "../common/strings";
...
+    {
+      mime: "image/tif",
+      pattern: [0x25, 0x50, 0x44, 0x46, 0x2d],
+    },
+    {
+      mime: "application/pdf",
+      pattern: [0x25, 0x50, 0x44, 0x46, 0x2d],
+    },
+    // We can expand this list @see https://mimesniff.spec.whatwg.org/#matching-an-image-type-pattern
+  ];
```
Comment (kunzms) [inline src/services/assetService.ts:54]: correct the indent

File: src/services/assetService.ts [hunk 7]
```diff
@@ -48,7 +84,20 @@ export class AssetService {
...
+
+        if (supportedImageFormats.includes(assetFormat)) {
+            const mime = await this.getMimeType(filePath);
+
+            // If file was renamed/spoofed - fix file extension to true MIME type and show message
+            if (assetFormat !== mime) {
+                assetFormat = mime;
+                const corruptFileName = fileName.split("%").pop().substring(2);
+                setTimeout(() => {
+                    toast.info(`Attention! ${corruptFileName.toLocaleUpperCase()} - extension of this file doesn't correspond MIME type. Please check file: ${corruptFileName.toLocaleUpperCase()}`, { autoClose: false });
```
Comment (kunzms) [inline src/services/assetService.ts:97]: Add localization strings

File: package.json (hunk 4 above)
Comment (alex-krasn) [reply package.json:22]: actually we are not using this librarary anymore - I'll delete it

File: src/services/assetService.ts (hunk 2 above)
Comment (alex-krasn) [reply src/services/assetService.ts]: not anymore

File: package.json (hunk 4 above)
Comment (alex-krasn) [reply package.json]: done

File: src/services/assetService.ts [hunk 8]
```diff
@@ -48,7 +91,20 @@ export class AssetService {
...
-        const assetFormat = extensionParts[0];
+        let assetFormat = extensionParts[0];
+
+        if (supportedImageFormats.hasOwnProperty(assetFormat)) {
+            const mime = await this.getMimeType(filePath);
+
+            // If file was renamed/spoofed - fix file extension to true MIME type and show message
+            if (assetFormat !== mime) {
+                assetFormat = mime;
+                const corruptFileName = fileName.split("%").pop().substring(2);
```
Comment (stew-ro) [inline src/services/assetService.ts:102]: Doesn't account for file names with spaces 
image

File: src/services/assetService.ts (hunk 5 above)
Comment (alex-krasn) [reply src/services/assetService.ts:26]: done

**Pull request merged**